### PR TITLE
feat(compiler): add milliseconds, days, months and years to duration

### LINKED ISSUE #2243: Add support for longer durations in `std.Duration` and duration literals
### Feature Spec

It is now possible to specify duration in days, months and years:

```js
let five_days = 5d;
let two_months = 2m;
let ten_years = 10y;
```

### Use Cases

Obvious

### Implementation Notes

_No response_

### Component

Language Design, SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by marciocadev**
Adding support to days, months and years to Duration

| syntax | description | in seconds |
|-|-|-|
| 1ms | 1 millisecond | 1/1000 |
| 1s | 1 second | 1 |
| 1m | 1 minute | 60 |
| 1h | 1 hour | 60 * 60 |
| 1d | 1 day | 60 * 60 * 24 |
| 1mo | 1 month | (60 * 60 * 24 * 365) / 12 ➡️ (30.4166667 days) |
| 1y | 1 year | 60 * 60 * 24 * 365 |

Months and years are a bit complicated as their duration varies, so I am assuming that 1 month (30,4166667 days) is equal to 2,592,000 seconds and 1 year corresponds to 31,104,000 seconds.

Closes #2243

## Checklist

- [x] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/tree-sitter-wing/grammar.js [hunk 1]
```diff
@@ -332,10 +332,13 @@ module.exports = grammar({
...
     bool: ($) => choice("true", "false"),
 
-    duration: ($) => choice($.seconds, $.minutes, $.hours),
+    duration: ($) => choice($.seconds, $.minutes, $.hours, $.days, $.months, $.years),
     seconds: ($) => seq(field("value", $.number), "s"),
     minutes: ($) => seq(field("value", $.number), "m"),
     hours: ($) => seq(field("value", $.number), "h"),
+    days: ($) => seq(field("value", $.number), "d"),
+    months: ($) => seq(field("value", $.number), "mo"),
+    years: ($) => seq(field("value", $.number), "y"),
```
Comment (hasanaburayyan) [inline libs/tree-sitter-wing/grammar.js:341]: I dont think we currently have duration expressions in our tree-sitter tests but maybe we should add them

**Review by hasanaburayyan — COMMENTED** [review]
🔥 I think we also need to add `fromDays()`, `fromMonths()`, and `fromYears()` in the std.Duration class as static methods.

**Comment by Chriscbr** [comment]
Can we also add milliseconds? (e.g. `123ms`) (may require slight tweak to the implementation of Duration in the code)

**Comment by marciocadev** [comment]
@hasanaburayyan and @Chriscbr 

I added tests for tree-sitter, the syntax `ms` to indicate milliseconds, and in `std.Duration`, the calls `fromDays()`, `fromMonths()`, `fromYears()`, and `fromMilliseconds()`.

File: libs/wingc/src/parser.rs [hunk 2]
```diff
@@ -133,6 +133,10 @@ impl<'s> Parser<'s> {
 		let value_text = self.node_text(&self.get_child_field(&value, "value")?);
 
 		match value.kind() {
+			"milliseconds" => Ok(Literal::Duration(
+				// Specific "Minutes" duration needed here
```
Comment (hasanaburayyan) [inline libs/wingc/src/parser.rs:137]:
```suggestion
```

**Review by hasanaburayyan — APPROVED** [review]

**Pull request merged**